### PR TITLE
Add a scraper to scrape disk usage metrics

### PR DESCRIPTION
**Pull request by tcnghia**

### CODE REVIEW

**Review by jonjohnsonjr — APPROVED** [review]

File: pkg/httpmetrics/disk_metrics.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	}
+	return usage
+}
+
+func ScrapeDiskUsage(ctx context.Context) {
+	once.Do(func() {
+		clog.FromContext(ctx).Info("Starting disk usage scraper with interval", "interval", DiskUsageScrapeInterval)
+
+		// Start a timer to scrape disk usage every 5 seconds.
+		ticker := time.NewTicker(DiskUsageScrapeInterval)
```
Comment (k4leung4) [inline pkg/httpmetrics/disk_metrics.go:73]: can we make this environment var configurable?

how often is our sidecar set to scrape?

File: pkg/httpmetrics/disk_metrics.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+// - and any additional volume mounts.
+//
+// We ignore all the /dev partitions here, keeping /, /tmp, and any additional
+// volumn mounts that may come from the Revision config.
+func scrapeDiskUsage() map[string]uint64 {
+	parts, err := disk.Partitions(true)
+	if err != nil {
+		// It is better to be silent here and missing metrics, than to be spam log
+		// here, and/or panic.
+		return nil
```
Comment (k4leung4) [inline pkg/httpmetrics/disk_metrics.go:49]: im fine not logging, 
can we add a metric for failures so we know how often its failing?

Comment (tcnghia) [reply pkg/httpmetrics/disk_metrics.go:58]: Done

**Review by k4leung4 — APPROVED** [review]

File: pkg/httpmetrics/disk_metrics.go (hunk 1 above)
Comment (tcnghia) [reply pkg/httpmetrics/disk_metrics.go]: Added an env var. Will remove later after we find a good default or confirm that 10s is a good default for all our usages.

**Pull request merged**